### PR TITLE
chore(changeset): baseBranch should the branch that PRs are based on / created against

### PR DESCRIPTION
**Pull request by sammoore**
This fixes an issue where users who just cloned the repo, and thus don't have the main branch locally (i.e. because the default branch is `beta`), are unable to create a changeset.

It's also the recommended approach according to the `changeset` documentation, which [Ernesto notes below](https://github.com/danstepanov/create-expo-stack/pull/79#issuecomment-1827605221)

Since this modifies the changeset configuration's baseBranch, it also affects the deployment process as outlined under the "Changes fro the Maintainer" heading.

## Changes for the Contributor

Before, contributors would need to have the `main` branch available locally in order to run `bunx changeset`. This wasn't automatic, because `main` is not the default branch.

After this PR, contributors should already have the `beta` branch available locally because it's the default branch (currently).
- **_In the future_**, if/when we change the `beta` branch name, an issue like this would occur again (for people who forked or cloned the project before the default branch name changed). To fix this, contributors who forked or cloned **before** the name change would need to update their fork (to set the new default branch) and checkout that branch locally.

## Changes for the Maintainer

As mentioned, I believe this changes the deployment process slightly, since changeset will autogenerate PRs against `beta` rather than `main` now. My understanding of the before/after:

### Before

1. Merge changes from `beta` (either with a PR or manually) to `main` when you want to do a release.
1. The `changeset` GitHub Actions will create (or update) a PR against `main`.
1. A maintainer merges that PR.
1. In order to have the changesets _removed_ from `.changeset/` (and the automated package.json version bump _present_) in the `beta` branch, the maintainer merges/pushes from `main` back into `beta` to get it caught back up. This should hopefully be a fast-forward merge (i.e. no merge commit) to keep the commit history clean.

### After

Differences from "Before" are in **bold**:

1. Merge changes from `beta` (either with a PR or manually) to `main` when you want to do a release.
1. The `changeset` GitHub Actions will create (or update) a PR against **`beta`**.
1. A maintainer merges that PR.
1. In order to have the changesets _removed_ from `.changeset/` (and the automated package.json version bump _present_) in the **`main`** branch, the maintainer merges/pushes from **`beta`** back into **`main`** to get it caught back up. This should hopefully be a fast-forward merge (i.e. no merge commit) to keep the commit history clean.

### The Differences

Compared to before, now:

1. After merging to `main`, the `changeset` GitHub Actions autogenerates the PR against `beta`, not `main`.
1. In the last step, the maintainer is updating `main` to have the changes that were just merged into `beta` -- not the other way around.

### CODE REVIEW

**Comment by sammoore** [comment]
To be sure I'm not totally in the wrong here, and that we don't break changeset again -- I'm requesting a review from @ernestoresende who set up changeset.

**Review by ernestoresende — APPROVED** [review]

**Comment by sammoore** [comment]
Upon further evaluation, this PR has one additional implication, as I believe it will change the GitHub Action for change-sets on the `main` branch to instead create a PR against `beta`/`develop`.

If we are okay with this workflow, I believe it will work like this after merging this PR:

1. Like before, merge to `main` when you want to do a release.
2. It will create a PR against `beta`/`develop`.
3. A maintainer merges said PR into `beta`/`develop`.
4. The maintainer merges/pushes (locally with the CLI, or however) back into `main` to get it caught back up (so the changesets and version bump are present in the `main` branch).

We could try to minimize manual intervention or change this around, but I think a stable, working procedure may be more important for now.

@danstepanov if you approve, we can merge this (so the likelihood of someone having an issue with `bunx changeset` is as small as possible) and I'll follow up with the name switch to `develop`.

**Comment by ernestoresende** [comment]
Delineated some more thoughts on a new RFC:
- https://github.com/danstepanov/create-expo-stack/discussions/86

But as it regards to this PR, the GitHub Action is hardcoded to run on a specific branch mentioned by name. The changes in this PR should not affect how the GitHub Actions run.

**Comment by sammoore** [comment]
> But as it regards to this PR, the GitHub Action is hardcoded to run on a specific branch mentioned by name. The changes in this PR should not affect how the GitHub Actions run.

Not quite. You're right that it's hardcoded to run against the `main` branch -- but the changeset GH action determines what branch to open it's _auto-generated pull request_  against depending on the `baseBranch`.

> Delineated some more thoughts on a new RFC

Checking this out now! 🙌

**Comment by danstepanov** [comment]
@sammoore what's the story on this PR?

**Comment by danstepanov** [comment]
image

**Pull request merged**
